backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/impl/ElasticsearchScopedIndexFieldComponent.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.search.impl;
+
+public class ElasticsearchScopedIndexFieldComponent<T> {
+
+	private final T component;
+	private final ElasticsearchConverterCompatibilityChecker compatibilityChecker;
+
+	public ElasticsearchScopedIndexFieldComponent(T component, ElasticsearchConverterCompatibilityChecker compatibilityChecker) {
+		this.component = component;
+		this.compatibilityChecker = compatibilityChecker;
+	}
+
+	public T getComponent() {
+		return component;
+	}
+
+	public ElasticsearchConverterCompatibilityChecker getCompatibilityChecker() {
```

Comment:
I'd use "ConverterCompatibilityChecker" because just "CompatibilityChecker" make it looks like it's about the whole field... It's confusing.